packages/amplify-cli/src/commands/upgrade.ts

```diff
@@ -25,32 +29,51 @@ export const run = async (context: $TSContext) => {
   }
   const latestVersion = await getLatestVersion();
   if (gt(latestVersion, thisVersion)) {
-    await upgradeCli(latestVersion);
+    await upgradeCli(context.print, latestVersion);
     context.print.success(`Successfully upgraded to Amplify CLI version ${latestVersion}!`);
   } else {
     context.print.info('This is the latest Amplify CLI version.');
   }
 };
 
-const upgradeCli = async (version: string) => {
+const upgradeCli = async (print, version: string) => {
   const isWin = process.platform.startsWith('win');
-  const binPath = path.join(pathManager.getHomeDotAmplifyDirPath(), 'bin', isWin ? 'amplify.exe' : 'amplify');
+  const binDir = path.join(pathManager.getHomeDotAmplifyDirPath(), 'bin');
+  const binPath = path.join(binDir, isWin ? 'amplify.exe' : 'amplify');
   const platformSuffix = isWin ? 'win.exe' : process.platform === 'darwin' ? 'macos' : 'linux';
-  const url = binUrl(version, platformSuffix);
-  const spinner = ora();
+  const extractedName = binName(platformSuffix);
+  const extractedPath = path.join(binDir, extractedName);
+  const url = binUrl(version, extractedName);
 
   if (isWin) {
     await fs.move(binPath, oldVersionPath);
   }
-
-  spinner.start('Downloading latest Amplify CLI version...');
   const response = await fetch(url);
   if (response.status >= 400) {
     throw new Error(`${response.status}: Request to ${url} failed:\n${JSON.stringify(response.json(), null, 2)}`);
   }
-  const bin = await response.buffer();
-  spinner.succeed('Download complete!');
-  await fs.writeFile(binPath, bin);
+  const len = response.headers.get('content-length');
```

Comment:
content-length is always guaranteed in http header right?

Reply:
not always, but GH download APIs include it.